daemon/container_unix.go

```diff
@@ -720,6 +720,10 @@ func (daemon *Daemon) updateNetworkSettings(container *Container, n libnetwork.N
 		container.NetworkSettings = &network.Settings{Networks: make(map[string]*network.EndpointSettings)}
 	}
 
+	if !container.hostConfig.NetworkMode.IsHost() && runconfig.NetworkMode(n.Type()).IsHost() {
+		return runconfig.ErrConflictSharedNetwork
```

Comment:
Both be `ErrConflictHostNetwork` is also make sense, just need to modify the description of `ErrConflictHostNetwork`, I'll update